### PR TITLE
Code refactoring

### LINKED ISSUE #53: Text wrapping doesn't account for margin
From what I can see there is no way currently to make this work:
Screenshot 2021-04-04 at 19 19 53

The problem is that our CLI includes a prefix for each line and `boxen` doesn't know that. I've wanted to include the margin in the options so I would just move the box to the right by the appropriate amount of characters (known to me). However, this doesn't work - since text wrapping etc happens before margin is even looked at.

### PR DESCRIPTION
**Pull request by Caesarovich**
This PR has for objective to provide a little code refactoring in order to make the library more easily maintainable/scalable/readable.
Effectively the main function of the library was around 150 lines long and it renders it really hard to read/understand.

So I decided to divide it a little, now the code is separated into two 'main functions' : 
- The main function from before without the content handling part.
- And the makeContentText function wich now handles the whole content part.

This new function takes in 4 arguments: 
- text: wich is the box's content text provided by the user
- padding: the optional padding desired
- columns: the maximum column count the content should take
- align: the optional text alignment

This allows for more flexibility regarding the width of the content. It'll allow an easy implementation for issue #51. (PR coming soon). Also the way this function is done/used fixes #53 and all kind of similar bugs where the box would exceed the terminal's size. See images below:

Image comparison of old/new version

This pretty much sums up the purpose of this PR. It's kind of a preparation for other PRs I'm preparing

---

Fixes #53

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -159,23 +198,19 @@ module.exports = (text, options) => {
...
-	}
+	text = makeContentText(text, padding, contentWidth, options.align);
 
-	const paddingLeft = PAD.repeat(padding.left);
 	let marginLeft = PAD.repeat(margin.left);
 
 	if (options.float === 'center') {
-		const padWidth = Math.max((columns - contentWidth - BORDERS_WIDTH) / 2, 0);
-		marginLeft = PAD.repeat(padWidth);
+		const marWidth = Math.max((columns - contentWidth - BORDERS_WIDTH) / 2, 0);
```
Comment (sindresorhus) [inline index.js:209]:
```suggestion
		const marginWidth = Math.max((columns - contentWidth - BORDERS_WIDTH) / 2, 0);
```

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by sindresorhus** [comment]
> Remove outdated tests that aren't relevant anymore

Why are the tests not relevant anymore?

**Comment by sindresorhus** [comment]
> Also the way this function is done/used fixes #53 and all kind of similar bugs where the box would exceed the terminal's size.

Would be nice to have tests for this.

**Comment by Caesarovich** [comment]
> > Remove outdated tests that aren't relevant anymore
> 
> Why are the tests not relevant anymore?

Those tests were expecting results that do not accord with the fact that the boxes shouldn't overflow the terminal.

I found that by investigating those and try running them in a separate script. I observed that the test produced a box exceeding the terminal's size and accounting it as Ok, while accounting the new version (The one that doesn't overflow anymore) as Ko.
It's due to the fact that the new way of doing the box I implemented works very differently than before in some ways, like cutting the text into newlines to not overflow, etc...

**Comment by Caesarovich** [comment]
> > Also the way this function is done/used fixes #53 and all kind of similar bugs where the box would exceed the terminal's size.
> 
> Would be nice to have tests for this.

This is a very good idea, I'll get to it.

**Comment by Caesarovich** [comment]
Changes done, waiting for your review 👍

**Comment by sindresorhus** [comment]
> Those tests were expecting results that do not accord with the fact that the boxes shouldn't overflow the terminal.

Wouldn't it be better to adjust the tests instead of removing them?

**Comment by Caesarovich** [comment]
Changes done, waiting for your review 

> > Those tests were expecting results that do not accord with the fact that the boxes shouldn't overflow the terminal.
> 
> Wouldn't it be better to adjust the tests instead of removing them?

I can do it. Tho the it is now equivalent as combining the 'align option' tests with the new 'overflow' test. If anything, one one these two would already Ko before.

I'll push it in the next minutes.

**Comment by Caesarovich** [comment]
Readapted the tests. I pushed the changes.  👌

File: test.js [hunk 2]
```diff
@@ -525,133 +525,162 @@ test('text is right-aligned after wrapping', t => {
...
 	const lines = [];
-	for (let index = 0; index < 4; ++index) {
+	for (let index = 1; index < 10; ++index) {
 		const line = box.slice(index * width, (index + 1) * width);
-		t.is(line.length, width, 'Length of line #' + index);
-		t.is(line, line.trim(), 'No margin of line #' + index);
+		lines.push(line);
+	}
 
+	lines.forEach(line => {
```
Comment (sindresorhus) [inline test.js:573]: I prefer using `for-of` instead of `.forEach`.

File: test.js [hunk 3]
```diff
@@ -525,133 +525,162 @@ test('text is right-aligned after wrapping', t => {
...
-		const line = box.slice(index * width, (index + 1) * width);
-		t.is(line.length, width, 'Length of line #' + index);
-		t.is(line, line.trim(), 'No margin of line #' + index);
-
-		lines.push(line);
-	}
+	const lines = box.split('\n');
 
-	const checkAlign = index => {
+	const checkAlign = (index, leftPad, rightPad) => {
```
Comment (sindresorhus) [inline test.js:674]:
```suggestion
	const checkAlign = ({index, leftPad, rightPad}) => {
```

To improve readability at the callsite.

**Comment by sindresorhus** [comment]
Looks good now. Thanks :)

**Pull request merged**